XiEditor/AppDelegate.swift

```diff
@@ -72,17 +51,28 @@ class AppDelegate: NSObject, NSApplicationDelegate {
             if let obj = params as? [String : AnyObject], let update = obj["update"] as? [String : AnyObject] {
                 guard let tab = obj["tab"] as? String
                     else { print("tab missing from update event"); return }
-                guard let appWindowController = appWindowControllers[tab]
-                    else { print("tab " + tab + " not registered"); return }
-                appWindowController.editView.updateSafe(update: update)
+                
+                //FIXME: a better way to get a document
```

Comment:
This is ok, but at least factor it out into a separate method to avoid duplication.